### ISSUE TITLE
from .cywinpty import Agent --> Error Python 3.6.4 and 2.7.x 

### ISSUE THREAD

**Issue by padremayi** [andfoy/pywinpty#93]
It is impossible to use the library due to this error:

> from .cywinpty import Agent
ImportError: DLL load failed: The specified procedure could not be found.

Executing
`python setup.py install`

I see this error but the procedure finishes without any problem:
`winpty/cywinpty.pyx: cannot find cimported module 'winpty._winpty'`

**Comment by andfoy** [comment]
@padremayi How did you installed the package?

**Comment by padremayi** [comment]
With
`python setup.py install`

**Comment by andfoy** [comment]
From which directory are you trying to perform the import?

**Comment by padremayi** [comment]
I wrote an application that uses pywinpty library. Simplifying I use the example in the main page:
```
# High level usage using `spawn`
from winpty import PtyProcess

proc = PtyProcess.spawn('python')
proc.write('print("hello, world!")\r\n')
proc.write('exit()\r\n')
while proc.isalive():
	print(proc.readline())
```

**Comment by andfoy** [comment]
But are you doing the tests outside your ``pywinpty`` clone folder?

**Comment by andfoy** [comment]
BTW, it is recommended to use ``pip install -U .``

**Comment by padremayi** [comment]
Yes, the error comes out during the first import:
```
C:\Users\MyPC\AppData\Local\Programs\Python\Python36-32\python.exe "C:/Users/MyPC/PycharmProjects/test_pywinpty/main.py"
Traceback (most recent call last):
  File "C:/Users/MyPC/PycharmProjects/test_pywinpty/main.py", line 1, in <module>
    from winpty import PtyProcess
  File "C:\Users\MyPC\AppData\Local\Programs\Python\Python36-32\lib\site-packages\pywinpty-0.5.1-py3.6-win32.egg\winpty\__init__.py", line 11, in <module>
    from .ptyprocess import PtyProcess
  File "C:\Users\MyPC\AppData\Local\Programs\Python\Python36-32\lib\site-packages\pywinpty-0.5.1-py3.6-win32.egg\winpty\ptyprocess.py", line 14, in <module>
    from .winpty_wrapper import PTY, PY2
  File "C:\Users\MyPC\AppData\Local\Programs\Python\Python36-32\lib\site-packages\pywinpty-0.5.1-py3.6-win32.egg\winpty\winpty_wrapper.py", line 13, in <module>
    from .cywinpty import Agent
ImportError: DLL load failed: Impossibile trovare il modulo specificato.
```

**Comment by andfoy** [comment]
@padremayi Please uninstall the package and install it using ``pip install -U .``

**Comment by padremayi** [comment]
I'll try right now and let you know.

Thanks

**Comment by padremayi** [comment]
Ok, in this way it works!

I executed:
`python setup.py build_ext -i --compiler=msvc`

then
`pip install -U pywinpty`

Is the first step necessary or not?

**Comment by padremayi** [comment]
Doing the same thing on another PC and... it doesn't work :-(

**Comment by padremayi** [comment]
I found the problem; sometimes on .\Lib\site-packages\winpty folder inside Python folder these files are missing:
winpty.dll
winpty-agent.exe
winpty-debugserver.exe

**Comment by andfoy** [comment]
@padremayi They are included by using the pip command instead of the usual ``python setup.py install``. We define additional packaging rules.

**Comment by andfoy** [comment]
I'm glad that is working now for you

**Comment by andfoy** [comment]
> I found the problem; sometimes on .\Lib\site-packages\winpty folder inside Python folder these files are missing:

If you install pywinpty by source, you must have installed winpty binaries beforehand

**Comment by padremayi** [comment]
Many thanks for your help on poiting me on the right direction. I confirm that with Python 3.4.x it doesn't work also compiling winpty from source.

Maybe when you resolve [the problem for Python 2.7.x](https://github.com/spyder-ide/pywinpty/issues/91#issuecomment-364125574) the fix will work also on Python 3.4.x
